app/scripts/services/dimStoreService.factory.js

```diff
@@ -874,6 +894,30 @@
 
       return quality;
     }
+    
+    function isItemNew(new_id) {
+        var allItems = [];
+
+        // all stores
+        _.each(_oldStores, function(store, id) {
+
+          // all armor in store
+          var items = _.filter(store.items, function(item) {
```

Comment:
You're looking for one item - IDs are unique (except for consumables, but forget those). What you could do is, in the beginning, instead of deep-copying the whole list of stores and everything in them, just build a list (or a map) of all the item IDs in all stores. Then just check if that list/map contains your item ids in this function.